### PR TITLE
fix: retry image pulling, stop on 404, no duplicate pulls

### LINKED ISSUE #2527: bump image pull timeout, option retry failures
## Feature Request

It's bad UX that Talos fails on image pull over slow connection.

### Description

1. Bump image pull timeout (for each operation).
2. Add an option to go-retry to print immediately (and de-duplicate) errors as encountered during the retry loop.
3. Bump retry timeout accordingly for allow several attempts for (1).

### PR DESCRIPTION
**Pull request by smira**
This uses go-retry feature
(https://github.com/talos-systems/go-retry/pull/3) to print errors being
retried.

If image is not found in the index, abort retries immediately.

Don't pull installer image twice (if already pulled by the validation
code before).

Fixes #2527 

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/2548)

### CODE REVIEW

File: go.mod
Comment (rsmitty) [inline go.mod:8]: PR looks fine, but we should probably wait until the other is merged from go-retry and fix this go mod.

File: internal/pkg/containers/image/image.go
Comment (Ulexus) [inline internal/pkg/containers/image/image.go:24]: nit: magic numbers

Comment (smira) [reply internal/pkg/containers/image/image.go:24]: any suggestions?  would you like to see it configurable? expose at constants?

**Review by rsmitty — APPROVED** [review]

**Review by Ulexus — APPROVED** [review]

**Pull request merged**
